__tests__/lifecycle-scripts.js

```diff
@@ -144,21 +144,15 @@ test('should inherit existing environment variables when setting via yarnrc', as
 });
 
 test('should not show any error messages when script ends successfully', async () => {
-  const stdout = await execCommand('test', 'script-success');
-
-  expect(stdout).toMatch(/Done in /);
+  await expect(execCommand('test', 'script-success')).resolves.toBeDefined();
 });
 
-test('should show correct error message when the script ends with an exit code', async () => {
-  await execCommand('test', 'script-fail').catch(error => {
-    expect(error.message).toMatch(/Command failed with exit code /);
-    expect(error.code).not.toBe(0);
-  });
+test('should throw error when the script ends with an exit code', async () => {
+  await expect(execCommand('test', 'script-fail')).rejects.toBeDefined();
```

Comment:
@BYK this test (also the test below) should fail when the script unexpectedly succeeds.